Make title of checkbox dropdown configurable

jbooherl opened the pull request:
Metrics would like to use the checkbox dropdown but with a different title. I believe the change is backwards compatible. If there is a better property name to use to prevent the warning from React about not recognizing a prop on a DOM element, please change the name.

sweinstein22 commented:
Hi @jbooherl , it would be great if you all could include changes to the `/styleguide` directory to document the change you're requesting. I'm also not sure what warning you're seeing about React not recognizing a prop on the DOM element, could you provide reproduction steps?

jbooherl commented:
Sure, I'll also make changes to the styleguide. Just remembered that this morning in stand up. I had originally named the property customizeTitle, but when I ran the jest tests I saw a React warning. I think it is related to this: https://reactjs.org/warnings/unknown-prop.html. I have not seen it before either.

jbooherl commented:
It looks like the prop warning was related to all the props being passed into the Dropdown element. I change the name to customizeTitle and removed it from the props passed to Dropdown. I also updated the styleguide.

sweinstein22 commented:
Hi @jbooherl, thanks for making the change and for adding documentation to the styleguide. For future PRs, it would be great if you could include an example of the change in use, especially if it is a new prop, so that users have a place to experiment with the property. I have made a commit to do so [here](https://github.com/pivotal-cf/pivotal-ui/commit/003c0f96f854679d17c0dff3ce508be8474a0934)

It would also be great if you could allow us to be the ones to merge in your proposed change to ensure we have time to do our due diligence on the changes you're requesting. I recognize that it can be frustrating to wait, especially as our team has many competing priorities, but we appreciate your patience and understanding in the matter.

We will try to work on cutting a patch release of Pivotal UI with your change in it sometime soon.
